src/server/controllers/DeviceController.ts

```diff
@@ -14,13 +19,14 @@ DeviceController.post('/', async (req, res) => {
 
     const { deviceId } = asQuery(req.query)
 
-    let device = await Device.fetch(deviceId)
-    if (device) {
-      await device.save(req.body)
+    let device = await fetchDevice(deviceId)
+    if (device != null) {
+      await saveDeviceToDB(asDevice(req.body))
```

Comment:
I thought that's what the `pushTypes` file is for? For example, the apiKey interface was defined in this file as an api type since it has a field "apiKey" that does not really exist on the database.